src/main/perl/ComponentProxyList.pm

```diff
@@ -776,14 +776,16 @@ sub set_state
 {
     my ($logger, $comp, $msg, $statedir, $noaction) = @_;
     if ($noaction) {
-        $msg = "needing to run" if (!$msg);
+        $msg = "needing to run" if ($msg ne '');
         $logger->info("would mark state of component $comp as '$msg' (noaction set)");
         return;
     }
 
+    my $msg_txt = $msg eq '' ? 'no message': "message: $msg";
+
```

Comment:
i wanted to make it clear that there is no message, otherwise people might think there is no message because we don't log the message.